core/arch/interp.c

```diff
@@ -3972,7 +3972,7 @@ build_bb_ilist(dcontext_t *dcontext, build_bb_t *bb)
     if (TEST(FRAG_HAS_RSEQ_ENDPOINT, bb->flags)) {
         instr_t *label = INSTR_CREATE_label(dcontext);
         instr_set_note(label, (void *)DR_NOTE_REG_BARRIER);
-        instrlist_meta_preinsert(bb->ilist, bb->instr, label);
+        instrlist_meta_append(bb->ilist, label);
```

Comment:
Worth adding an assert that bb->instr is null?

Reply:
I think the preinsert was a mistake and we want to postinsert/append regardless to ensure the barrier is after the full 1st instru run of the rseq sequences and right before the 2nd native run.  Added a comment.